src/basic/socket-util.c

```diff
@@ -1226,3 +1241,142 @@ int socket_pass_pktinfo(int fd, bool b) {
                 return -EAFNOSUPPORT;
         }
 }
+
+int socket_pass_err(int fd, int af, bool b) {
+        int r;
+
+        if (af == AF_UNSPEC) {
+                r = socket_get_family(fd, &af);
+                if (r < 0)
+                        return r;
+        }
+
+        switch (af) {
+
+        case AF_INET:
+                return setsockopt_int(fd, IPPROTO_IP, IP_RECVERR, b);
+
+        case AF_INET6:
+                return setsockopt_int(fd, IPPROTO_IPV6, IPV6_RECVERR, b);
+
+        default:
+                return -EAFNOSUPPORT;
+        }
+}
+
+int socket_pass_ttl(int fd, int af, bool b) {
```

Comment:
Hmm, why is this called "pass_ttl" not "set_ttl"? Oh, OK, because there's already a set_ttl variant. Yeah, please name the function after the socket option. (And if there's different naming for ipv4 and ipv6, just pick the name that makes more sense.)